packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -183,6 +184,50 @@ interface JavaProp {
 }
 
 class JavaGenerator extends Generator {
+    // When the code-generator needs to generate code for a property or method that has the same name as a member of this list, the name will
+    // be automatically modified to avoid compile errors. Most of these are java language reserved keywords. In addition to those, any keywords that
+    // are likely to conflict with auto-generated methods or properties (eg: 'build') are also considered reserved.
+    private static RESERVED_KEYWORDS = [
+        'abstract', 'assert', 'boolean', 'break', 'build', 'byte', 'case', 'catch', 'char', 'class',
+        'const', 'continue', 'default', 'double', 'do', 'else', 'enum', 'extends', 'false',
+        'final', 'finally', 'float', 'for', 'goto', 'if', 'implements', 'import', 'instanceof',
+        'int', 'interface', 'long', 'native', 'new', 'null', 'package', 'private', 'protected',
+        'public', 'return', 'short', 'static', 'strictfp', 'super', 'switch', 'synchronized',
+        'this', 'throw', 'throws', 'transient', 'true', 'try', 'void', 'volatile', 'while'
+    ];
+
+    /**
+     * Turns a raw javascript property name (eg: 'default') into a safe Java property name (eg: 'defaultValue').
+     * @param propertyName the raw JSII property Name
+     */
+    private static safeJavaPropertyName(propertyName: string) {
+        if (!propertyName) {
+            return propertyName;
+        }
+
+        if (JavaGenerator.RESERVED_KEYWORDS.includes(propertyName)) {
+            return `${propertyName}Value`;
```

Comment:
Might be worth checking whether this is the name of an existing property on the same type... If that is so, just bail out saying you're out of options there...

Reply:
Difficult to do, so in the interests of unblocking CDK builds moving ahead without this functionality for now.